server/reload.go

```diff
@@ -1100,6 +1130,24 @@ func (s *Server) reloadClientTraceLevel() {
 	}
 }
 
+// reloadClusterName detects cluster membership changes triggered
+// due to a reload where the name changes.
+func (s *Server) reloadClusterName() {
+	s.mu.Lock()
+	// Get all connected routes and notify the cluster rename.
+	infoJSON := s.routeInfoJSON
+	for _, route := range s.routes {
+		route.mu.Lock()
+		route.enqueueProto(infoJSON)
+		route.mu.Unlock()
+	}
+
+	// Reloading without an explicit cluster name makes a server
+	// be susceptible to dynamic cluster name changes.
+	s.susceptible = s.opts.Cluster.Name == ""
```

Comment:
Just a note that it is safe here to use `s.opts.` directly since we are in the "thread" that does config reload and the options have just been set, but otherwise I recommend using s.getOpts().